src/driver/test_point_aggregator.py

```diff
@@ -0,0 +1,130 @@
+    ),
+    "relative_error_q30": lambda x, y: np.quantile(
+        np.abs(_reach(x) - _reach(y)) / _reach(x), 0.30
+    ),
+    "relative_error_q40": lambda x, y: np.quantile(
+        np.abs(_reach(x) - _reach(y)) / _reach(x), 0.40
+    ),
+    "relative_error_q50": lambda x, y: np.quantile(
+        np.abs(_reach(x) - _reach(y)) / _reach(x), 0.50
+    ),
+    "relative_error_q60": lambda x, y: np.quantile(
+        np.abs(_reach(x) - _reach(y)) / _reach(x), 0.60
+    ),
+    "relative_error_q70": lambda x, y: np.quantile(
+        np.abs(_reach(x) - _reach(y)) / _reach(x), 0.70
+    ),
+    "relative_error_q80": lambda x, y: np.quantile(
+        np.abs(_reach(x) - _reach(y)) / _reach(x), 0.80
+    ),
+    "relative_error_q90": lambda x, y: np.quantile(
+        np.abs(_reach(x) - _reach(y)) / _reach(x), 0.90
+    ),
+    # Mean shuffle distance
+    "mean_shuffle_distance": lambda x, y: np.mean(
+        [_shuffle_distance(x[i], y[i]) for i in range(len(x))]
+    ),
+    # Mean squared shuffle distance
+    "mean_squared_shuffle_distance": lambda x, y: np.mean(
+        [_shuffle_distance(x[i], y[i]) ** 2 for i in range(len(x))]
+    ),
+    # Variance of shuffle distance
+    "var_shuffle_distance": lambda x, y: np.var(
+        [_shuffle_distance(x[i], y[i]) for i in range(len(x))]
+    ),
+}
+
+
+def _reach(point_list: List[ReachPoint], k=1) -> np.array:
+    """Returns list of k+ frequencies from list of ReachPoints."""
+    return np.array([point.reach(k) for point in point_list])
+
+
+def _shuffle_distance(xpoint: ReachPoint, ypoint: ReachPoint, k=5) -> float:
+    """Computes shuffle distance of first k frequency buckets."""
+    if xpoint.max_frequency <= k or ypoint.max_frequency <= k:
+        return 1.0
+    xfreq = np.array([xpoint.frequency(i + 1) for i in range(k)])
+    yfreq = np.array([ypoint.frequency(i + 1) for i in range(k)])
+    if sum(xfreq) == 0 or sum(yfreq) == 0:
+        return 0.5
+    return 0.5 * np.sum(np.abs(xfreq / sum(xfreq) - yfreq / sum(yfreq)))
+
+
+def aggregate(
+    true_reach: List[ReachPoint], simulated_reach: List[ReachPoint]
+) -> pd.DataFrame:
+    """Returns a DataFrame of the statistics listed in keys.
+
+    Args:
+      keys:  A list of strings.  Each string should specify the name of an aggregation
+        statistic, as given in AGGREGATORS.
+      true_reach:  A list of points representing true reach values.
+      simulated_reach:  A list of points representing modeled reach values.  This list must
+        be of the same length as true_reach.  The value of simulated_reach[i] should be the
+        output of the modeling function for the spend vector that was used to compute
+        true_reach[i].
+    Returns:
+      A single row DataFrame representing the values of the statistics listed in keys.
+    """
+    stats = {}
+    for key in AGGREGATORS:
+        stats[key] = [AGGREGATORS[key](true_reach, simulated_reach)]
+    return pd.DataFrame(data=stats)
```

Comment:
This returns a one-row dataframe with index 0. If we later combine (append) the metrics for N different reach points, we obtain an N-row dataframe with every row index being 0. 

It might be desirable to distinguish each reach point by row index. One option is to add an arg `name` in this method and return `pd.DataFrame(data=stats, index=[name])`. Not sure if it's useful!

Reply:
I will keep that in mind for later.  Thanks for the suggestion!